subset/connection/mac_oui/src/main/java/RetrieveList.java

```diff
@@ -1,29 +1,40 @@
-import java.io.BufferedReader;
-import java.io.InputStream;
-import java.io.InputStreamReader;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
 import java.util.HashMap;
 import java.util.Map;
 
 public class RetrieveList {
   String macAddress;
   Map<String, String> macDevices = new HashMap<>();
   static final int minimumMACAddressLength = 5;
+  private static final int BUFFER_SIZE = 4096;
 
   RetrieveList(String macAddress) {
     this.macAddress = macAddress;
   }
 
   public void startTest() {
-    readLocalFile();
-    System.out.println(String.format("Found %d entries in mac table\n", macDevices.size()));
+    try {
+      // Read the mac prefixes from the svn nmap
+      readFileFromUrlAndUpdateMac("https://svn.nmap.org/nmap/nmap-mac-prefixes");
```

Comment:
this should not do this at runtime -- this file should be downloaded when the Docker image is created, and then accessed here as a static file -- which is already done in Docker.test_macoui

Reply:
Yes there is. I can remove this as it is redundant now.